setup_env.bat

```diff
@@ -1,35 +1,54 @@
 @echo off
-cls
-set dirpath=%~dp0
-if "%dirpath:~-1%" == "\" set dirpath=%dirpath:~0,-1%
 
-git --version > nul
+REM Get the directory path of the script
+set "dirpath=%~dp0"
+if "%dirpath:~-1%" == "\" set "dirpath=%dirpath:~0,-1%"
+
+REM Check if git is installed
+git --version > nul 2>&1
 if %errorlevel% NEQ 0 (
     echo No git executable found in PATH!
     echo:
     pause
-    exit
+    exit /b 1
 )
 
+REM Create the virtual environment if it doesn't exist
 if not exist "%dirpath%\env" (
     echo:
     echo Creating the env folder...
     python -m venv "%dirpath%\env"
     if %errorlevel% NEQ 0 (
         echo:
-        echo No python executable found in PATH!
+        echo No python executable found in PATH or failed to create virtual environment!
         echo:
         pause
+        exit /b 1
     )
 )
 
+REM Activate the virtual environment and install requirements
 echo:
 echo Installing requirements.txt...
 "%dirpath%\env\scripts\python" -m pip install -U pip
+if %errorlevel% NEQ 0 (
+    echo Failed to upgrade pip.
+    exit /b 1
+)
+
```

Comment:
would it be best to just remove the whole exit line for both? so it still echos that it fails but doesn't exit

Reply:
Honestly: I don't really know. Either of the solutions is okay by me. The only thing I can see happening here, is that if the installation fails, and if there's anything fail-related displayed to the user, they may get confused and try to report it, which as explained, isn't really that big of a deal. Also, the first installation failing would mean the `wheel` installation, as well as the rest of the requirements, will also most likely fail, but that's something we can't really predict or guarantee would happen, so...

If you're unsure, due to the above, it'd probably be best not to print anything fail-related to the console.

Reply:
Oh do you also want me to remove it from the requirements part? since that would fail too

Reply:
The `requirements.txt` failing to install is an actual problem, so it can stay there. The failure would be obvious from the command output already though, so... again, not entirely sure. Up to you as well.

Reply:
Perfect, then i'll just let it be, everything should be ready then